src/shared/utilities/timeoutUtils.ts

```diff
@@ -39,6 +47,13 @@ export class Timeout {
         return remainingTime > 0 ? remainingTime : 0
     }
 
+    /**
+     * True when the Timeout has completed
+     */
+    public get fulfilled(): boolean {
+        return this._fulfilled
```

Comment:
The docstring is not really true or at least confusing, or conflicts with the docstring for timer().  The Timeout "completes" when the time runs out. Whereas `fulfilled()`  indicates exactly one thing: "killTimer() was called".

I suggest changing the docstring to be more explicit: `/** True if killTimer() was called */`
, since anything else is confusing because the Timeout class is gaining scope and its semantics are not clear...

Reply:
`this._fulfilled` is set to true when the promise resolves or rejects. Thus it is guaranteed that `fulfilled` is true when `timerPromise` has been fulfilled (either through `killTimer` or the `setTimeout` in the constructor).